Adds setup files

StefanD986 opened the pull request:
I tried to use the state of the art as described by the PyPA. Additionally I also used setuptools_scm to auto-generate version info. Additional benefit is that is simplifies package data management, so no need to maintain a manifest (not applicable here, because AFAIK there is no package data).

To build sdist and bdist wheels run install https://pypi.org/project/build/ and then run `python -m build --sdist --wheel .`

To version the package create tags on the master branch. (see setuptools_scm documentation for details)

I filled in the metadata to the best of my knowledge.

@luciano-fiandesio Let me know if this is useful or if you have questions.

StefanD986 committed:
Adds setup files

I tried to use the state of the art as described by the PyPA. Additionally I also used setuptools_scm to auto-generate version info. Additional benefit is that is simplifies package data management, so no need to maintain a manifest (not applicable here, because AFAIK there is no package data)

To build sdist and bdist wheels run install https://pypi.org/project/build/ and then run `python -m build --sdist --wheel .`

To version the package create tags on the master branch. (see setuptools_scm documentation for details)

luciano-fiandesio commented on setup.cfg:
```diff
@@ -0,0 +1,14 @@
+[metadata]
+name = beanborg
+url = https://github.com/luciano-fiandesio/beanborg
+author = Luciano Fiandesco
```
Small typo: Fiandesco -> Fiandesio

StefanD986 replied:
Fixed

luciano-fiandesio reviewed (approved):
Thanks, I was planning to add some sort of setup, but didn't get the time. What would be the next step to be able to install beanborg using `pip`?

StefanD986 committed:
Add console entry points to path

specifying the console entry points here will put the beanborg commands on the users PATH after installing the package.

StefanD986 committed:
Fix ModuleNotFoundError

When the package is pip-installed the package local imports must be qualified with the package name.

StefanD986 commented:
> Thanks, I was planning to add some sort of setup, but didn't get the time. What would be the next step to be able to install beanborg using `pip`?

pip installing should already be possible on this branch:
```bash
pip install git+https://github.com/StefanD986/beanborg.git@add-setup-files
```

Or - once this is merged to your master - you can also leave out the `@add-setup-files` and use your repo.
```
pip install git+https://github.com/luciano-fiandesio/beanborg.git
```

If you want this to be installable from PyPI, further steps would be 
  - check requirements for packages that are hosted on PyPI (they require certain metadata IIRC)
  - tag a release (create an annotated tag on the master branch e.g. `git tag -a v1.0`)
  - create a release wheel and/or sdist `python -m build --sdist --wheel .`
  - upload these to PyPI, either manually or maybe set up some automatic build with github actions and have an automatic upload. I guess depends on how often you plan to release.
